Added support for sts assume role

zoltrain opened the pull request:
- swapped out the goamz for the official aws sdk
  - supports loading config from the credentials file, env vars and
    through assume role.
  - supports mfa authentication against assume role
  - caches the credentiasl in ~/.aws/cli/cached/

Commit:
Added support for sts assume role

 - swapped out the goamz for the official aws sdk
 - supports loading config from the credentials file, env vars and
   through assume role.
 - supports mfa authentication against assume role
 - caches the credentiasl in ~/.aws/cli/cached/

dwb commented on aws.go:
```diff
@@ -1,96 +1,210 @@
+	var err error
+	for {
+		fmt.Print("Enter MFA code: ")
+		fmt.Scanln(&token)
```
This ignores the error possibly returned

Commit:
Merge branch 'master' into wip-added-assume-role-support

Commit:
Added support for non MFA assumption of sts role

Commit:
Shared cached creds with the AWS CLI

 - honoured expiry correctly, can't use StaticCreds IsExpired as this
   always returns false, have to do the evaluation manually

Commit:
Catch potential error reading from stdin

Commit:
Fixed issue of no throwing an error when a source_profile isn't found

 - when looking a source profile for assuming a role, the second profile
   lookup might fail. Added raising an error in this case

Commit:
Fixed issue with project name from env overwriting file based profile

 - when reading the AWS_DEFAULT_PROFILE env var is was last to be
   evaluated meaning it could blank a value read from the directory.

Commit:
Fixed bug with no creating the MFA cache path first

Commit:
Added default region to profile

 - make the project flag highest priority when loading profiles
